src/main/webapp/app/shared/util/core-submission-shared/core-submission-utils.ts

```diff
@@ -1,33 +1,42 @@
 import { Review } from 'app/shared/model/firebase/firebase.model';
 
-export function useLastReviewedOnly<T>(obj: T): T | undefined {
-  function useLastReviewedOnlyRec(curObj: T): T | undefined {
+function shouldProtect(valuePath: string | undefined, path: string) {
+  return valuePath !== undefined && path.length > 0 && (path.startsWith(valuePath) || valuePath.startsWith(path));
+}
+
+export function useLastReviewedOnly<T>(obj: T, valuePath?: string): T | undefined {
+  function useLastReviewedOnlyRec(curObj: T, parentPath: string): T | undefined {
     const newObj: T = {} as T;
     for (const [key, value] of Object.entries(curObj as Record<string, unknown>)) {
+      const currentPath = `${parentPath}${parentPath.length > 0 ? '/' : ''}${key}`;
+      const protectCurrent = shouldProtect(valuePath, currentPath);
+      const protectParent = shouldProtect(valuePath, parentPath);
+
       if (key.endsWith('_review')) {
         const temp: Review = Object.assign({} as Review, value);
-        if (temp?.added) {
-          return undefined;
-        }
         (newObj as Record<string, unknown>)[key] = temp;
       } else if (Array.isArray(value)) {
         const reviewObj: Review | undefined = curObj[`${key}_review`] as Review;
-        if (reviewObj?.added) {
+        if (reviewObj?.added && !protectParent) {
           return undefined;
         }
-        const arr: T[] = [];
+        const arr: (T | undefined)[] = [];
+        let i = 0;
         for (const arrObj of value) {
-          const temp = useLastReviewedOnlyRec(arrObj);
-          if (temp !== undefined) {
+          const temp = useLastReviewedOnlyRec(arrObj, `${currentPath}/${i}`);
+          if (temp !== undefined || protectParent) {
```

Comment:
keep undefined value in the array so that the index still exists for a path like `mutation/11/....`